packages/api-explorer/src/Doc.jsx

```diff
@@ -70,69 +70,56 @@ class Doc extends React.Component {
+    this.Params = createParams(this.props.oas, this.operation);
 
     this.resetForm = this.resetForm.bind(this);
 
     this.toggleAuth = this.toggleAuth.bind(this);
     this.waypointEntered = this.waypointEntered.bind(this);
-  }
-
-  componentDidMount() {
-    if (!this.shouldEnableRequestBodyJsonEditor()) {
-      return;
-    }
 
-    this.operation
-      .getRequestBodyExamples()
-      .then(examples => {
-        if (!Object.keys(examples).length) {
-          return;
-        }
+    // If we need to build out the request body JSON editor let's pull examples to fill it with.
+    if (this.shouldEnableRequestBodyJsonEditor()) {
+      const examples = this.operation.getRequestBodyExamples();
+      if (!Object.keys(examples).length) {
+        return;
+      }
 
-        const jsonExamples = examples.filter(
-          ex => matchesMimeType.json(ex.mediaType) || matchesMimeType.wildcard(ex.mediaType)
-        );
+      const jsonExamples = examples.filter(
+        ex => matchesMimeType.json(ex.mediaType) || matchesMimeType.wildcard(ex.mediaType)
+      );
 
-        if (jsonExamples.length) {
-          const example = jsonExamples[0];
-          let code = false;
+      if (jsonExamples.length) {
+        const example = jsonExamples[0];
+        let code = false;
 
-          if (example.code) {
-            code = example.code;
-          } else if (example.multipleExamples) {
-            code = example.multipleExamples[0].code;
-          }
+        if (example.code) {
+          code = example.code;
+        } else if (example.multipleExamples) {
+          code = example.multipleExamples[0].code;
+        }
 
-          try {
-            // Examples are stringified when we get them from `oas` because they need to be stringified for
-            // `@readme/syntax-highlighter` but because we need to pass a usable non-stringified object/array/primitive
-            // to our `CodeSample` component and `@readme/oas-to-har` we're parsing it out here. Cool? Cool.
-            code = JSON.parse(code);
-          } catch (e) {
-            code = {};
-          }
+        try {
+          // Examples are stringified when we get them from `oas` because they need to be stringified for
+          // `@readme/syntax-highlighter` but because we need to pass a usable non-stringified object/array/primitive
+          // to our `CodeSample` component and `@readme/oas-to-har` we're parsing it out here. Cool? Cool.
+          code = JSON.parse(code);
+        } catch (e) {
+          code = {};
+        }
 
-          code = code || {};
+        code = code || {};
 
-          this.setState({
-            formDataJson: code,
-            formDataJsonOriginal: code,
-            formDataJsonRaw: stringifyPretty(code),
-          });
-        }
-      })
-      .catch(() => {
-        // If we fail to generate examples for whatever reason fail silently.
-      });
+        this.state.formDataJson = code;
```

Comment:
So we can attach it straight onto the state now cos it's happening synchronously?

Reply:
Yep, it's no longer a promise so we don't need to do all this crazy handling.